ompi/mpi/c/info_set.c

```diff
@@ -97,6 +97,17 @@ int MPI_Info_set(MPI_Info info, const char *key, const char *value)
         }
     }
 
+// An extra warning condition is a key that uses our reserved prefix "__IN_".
+// That one is used internally to deal with the dynamic nature the key/val
+// pairs where we have callbacks that modify the val, and the MPI standard
+// wants the get_info call to give back the original setting rather than
+// the callback-modified setting. So if a user directly used a key __IN_foo
+// it would confuse our accounting slightly.
+    if (0 == strncmp(key, "__IN_", 5)) {
```

Comment:
Given that you use the string `"__IN_"` in multiple places in the code base, it might be good to either `#define` it, or have a global const string somewhere, and have all places in the code where you currently have `"__IN_"` use that global reference, instead.

Should we have the word `OMPI` in there somewhere?  E.g., `_OMPI_IN_`?  I only ask because it's a little weird to warn users about not naming info keys `__IN_`.